### PR TITLE
Add internal builder for service properties.

### LINKED ISSUE #11351: 2.1 preview: System.InvalidOperationException: 'The service property [...] of type [...] cannot be added to the entity type [...] because service property [...] of the same type already exists on entity type [...].'
I'm using the 2.1 preview.

My entities derive from a base class `EFBOBase`. This class has multiple properties of type `ContextBase`, and my context derives from `ContextBase`, which in turn derives from `DbContext`. These properties are not mapped and not in any way relevant to EF - my mapping contains `builder.Ignore()` calls for them. However, the exception happens before my `OnModelCreating` is even called - EF seems to interpret these properties as "service properties".

`System.InvalidOperationException: 'The service property 'ContextTestOne' of type 'ContextBase' cannot be added to the entity type 'AbsDisposition' because service property 'ContextTestTwo' of the same type already exists on entity type 'AbsDisposition'.'`

The line in my code that sets off the exception is `ChangeTracker.AutoDetectChangesEnabled = false;`

```
 	mscorlib.dll!System.Lazy<Microsoft.EntityFrameworkCore.Metadata.IModel>.CreateValue()	Unknown	Non-user code. Skipped loading symbols.
 	mscorlib.dll!System.Lazy<Microsoft.EntityFrameworkCore.Metadata.IModel>.LazyInitValue()	Unknown	Non-user code. Skipped loading symbols.
 	mscorlib.dll!System.Lazy<Microsoft.EntityFrameworkCore.Metadata.IModel>.Value.get()	Unknown	Non-user code. Skipped loading symbols.
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.GetModel(Microsoft.EntityFrameworkCore.DbContext context = {PI.PI_STD.DataAccess.Generated.Context.Context}, Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.IConventionSetBuilder conventionSetBuilder = {Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.CompositeConventionSetBuilder}, Microsoft.EntityFrameworkCore.Infrastructure.IModelValidator validator = {Microsoft.EntityFrameworkCore.Internal.OracleModelValidator})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Internal.DbContextServices.CreateModel()	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Internal.DbContextServices.Model.get()	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.Infrastructure.EntityFrameworkServicesBuilder.TryAddCoreServices.AnonymousMethod__7_1(System.IServiceProvider p = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(Microsoft.Extensions.DependencyInjection.ServiceLookup.FactoryCallSite factoryCallSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.FactoryCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope scope = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope, object>.VisitCallSite(Microsoft.Extensions.DependencyInjection.ServiceLookup.IServiceCallSite callSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.FactoryCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope argument = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(Microsoft.Extensions.DependencyInjection.ServiceLookup.ScopedCallSite scopedCallSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ScopedCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope scope = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope, object>.VisitCallSite(Microsoft.Extensions.DependencyInjection.ServiceLookup.IServiceCallSite callSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ScopedCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope argument = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite constructorCallSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope scope = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope, object>.VisitCallSite(Microsoft.Extensions.DependencyInjection.ServiceLookup.IServiceCallSite callSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope argument = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(Microsoft.Extensions.DependencyInjection.ServiceLookup.ScopedCallSite scopedCallSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ScopedCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope scope = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope, object>.VisitCallSite(Microsoft.Extensions.DependencyInjection.ServiceLookup.IServiceCallSite callSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ScopedCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope argument = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(Microsoft.Extensions.DependencyInjection.ServiceLookup.IServiceCallSite callSite = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ScopedCallSite}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope scope = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.RealizeService.AnonymousMethod__0(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope scope = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(System.Type serviceType = {Name = "IDbContextDependencies" FullName = "Microsoft.EntityFrameworkCore.Internal.IDbContextDependencies"}, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope serviceProviderEngineScope = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(System.Type serviceType = {Name = "IDbContextDependencies" FullName = "Microsoft.EntityFrameworkCore.Internal.IDbContextDependencies"})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.Abstractions.dll!Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(System.IServiceProvider provider = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope}, System.Type serviceType = {Name = "IDbContextDependencies" FullName = "Microsoft.EntityFrameworkCore.Internal.IDbContextDependencies"})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.Extensions.DependencyInjection.Abstractions.dll!Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService<Microsoft.EntityFrameworkCore.Internal.IDbContextDependencies>(System.IServiceProvider provider = {Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope})	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.DbContext.DbContextDependencies.get()	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.DbContext.InternalServiceProvider.get()	Unknown	Non-user code. Cannot find or open the PDB file.
 	Microsoft.EntityFrameworkCore.dll!Microsoft.EntityFrameworkCore.DbContext.ChangeTracker.get()	Unknown	Non-user code. Cannot find or open the PDB file.
>	PI.Lib.DataAccess.dll!PI.Lib.DataAccess.EF.ContextBase.ContextBase(System.Data.Common.DbConnection connection = {Oracle.DataAccess.Client.OracleConnection}) Line 104	C#	Symbols loaded.
 	PI.PI_STD.DataAccess.dll!PI.PI_STD.DataAccess.Generated.Context.Context.Context(System.Data.Common.DbConnection con = {Oracle.DataAccess.Client.OracleConnection}) Line 33	C#	Symbols loaded.
 	PI.PI_STD.DataAccess.dll!PI.PI_STD.DataAccess.EF.ContextFactory.CreateContext(out PI.Lib.DataAccess.EF.ContextTicket ticket = null, string schemaName = "") Line 40	C#	Symbols loaded.
 	EfTest.exe!EfTest.Program.Main(string[] args = {string[0]}) Line 31	C#	Symbols loaded.
```

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Make Ignore, conventions, properties and navigations treat service properties consistently.
Break up EntityTypeTest

Fixes #11351

### CODE REVIEW

File: src/EFCore/Metadata/Conventions/Internal/ServicePropertyDiscoveryConvention.cs [hunk 1]
```diff
@@ -13,11 +17,21 @@ namespace Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal
...
+        IModelBuiltConvention
     {
         private readonly ITypeMappingSource _typeMappingSource;
         private readonly IParameterBindingFactories _parameterBindingFactories;
 
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        private const string DuplicateServicePropertiesAnnotationName = "RelationshipDiscoveryConvention:DuplicateServiceProperties";
```
Comment (smitpatel) [inline src/EFCore/Metadata/Conventions/Internal/ServicePropertyDiscoveryConvention.cs]: private doc comments?

File: src/EFCore/Metadata/Conventions/Internal/ServicePropertyDiscoveryConvention.cs [hunk 2]
```diff
@@ -42,35 +56,58 @@ public virtual InternalEntityTypeBuilder Apply(InternalEntityTypeBuilder entityT
...
-                var candidates = entityType.ClrType.GetRuntimeProperties();
+                return entityTypeBuilder;
+            }
+
+            var candidates = entityType.ClrType.GetRuntimeProperties();
 
-                foreach (var propertyInfo in candidates)
+            foreach (var propertyInfo in candidates)
+            {
+                if (entityTypeBuilder.IsIgnored(propertyInfo.Name, ConfigurationSource.Convention)
```
Comment (smitpatel) [inline src/EFCore/Metadata/Conventions/Internal/ServicePropertyDiscoveryConvention.cs]: Can Convention ignore a property?

Comment (AndriySvyryd) [reply src/EFCore/Metadata/Conventions/Internal/ServicePropertyDiscoveryConvention.cs]: It can, but I don't think we have any that do.

Comment (smitpatel) [reply src/EFCore/Metadata/Conventions/Internal/ServicePropertyDiscoveryConvention.cs]: Condition should be unnecessary then

Comment (AndriySvyryd) [reply src/EFCore/Metadata/Conventions/Internal/ServicePropertyDiscoveryConvention.cs]: The condition is: "Is this ignored at convention source or higher?", so I could replace it with `entityTypeBuilder.IsIgnored(propertyInfo.Name, ConfigurationSource.DataAnnotations)`, but I don't think it's much of a difference

**Pull request merged**